### ISSUE TITLE
Unknown handler async in SG6-beta1 plugin when disabling basicauth

### ISSUE THREAD

**Issue by mikn** [floragunncom/search-guard-kibana-plugin#49]
```
{"type":"log","@timestamp":"2017-12-08T04:40:33Z","tags":["status","plugin:kibana@6.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-12-08T04:40:33Z","tags":["status","plugin:elasticsearch@6.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-12-08T04:40:34Z","tags":["status","plugin:searchguard@6.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Search Guard HTTP Basic Authentication is disabled.","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-12-08T04:40:34Z","tags":["status","plugin:searchguard@6.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - Search Guard multitenancy disabled","prevState":"yellow","prevMsg":"Search Guard HTTP Basic Authentication is disabled."}
{"type":"log","@timestamp":"2017-12-08T04:40:34Z","tags":["status","plugin:searchguard@6.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - Search Guard copy JWT params disabled","prevState":"yellow","prevMsg":"Search Guard multitenancy disabled"}
{"type":"log","@timestamp":"2017-12-08T04:40:34Z","tags":["fatal"],"pid":1,"level":"fatal","message":"Unknown handler: async","error":{"message":"Unknown handler: async","name":"Error","stack":"Error: Unknown handler: async\n    at Object.exports.assert (/usr/share/kibana/node_modules/hoek/lib/index.js:736:11)\n    at Object.exports.defaults (/usr/share/kibana/node_modules/hapi/lib/handler.js:108:14)\n    at new module.exports.internals.Route (/usr/share/kibana/node_modules/hapi/lib/route.js:60:37)\n    at internals.Connection._addRoute (/usr/share/kibana/node_modules/hapi/lib/connection.js:387:19)\n    at internals.Connection._route (/usr/share/kibana/node_modules/hapi/lib/connection.js:379:18)\n    at internals.Plugin._apply (/usr/share/kibana/node_modules/hapi/lib/plugin.js:572:14)\n    at internals.Plugin.route (/usr/share/kibana/node_modules/hapi/lib/plugin.js:542:10)\n    at exports.default (/usr/share/kibana/plugins/searchguard/lib/configuration/routes/routes.js:21:10)\n    at ScopedPlugin.init [as externalInit] (/usr/share/kibana/plugins/searchguard/index.js:241:13)\n    at ScopedPlugin.tryCatcher (/usr/share/kibana/node_modules/bluebird/js/main/util.js:26:23)\n    at Promise.attempt.Promise.try (/usr/share/kibana/node_modules/bluebird/js/main/method.js:30:24)\n    at /usr/share/kibana/src/server/plugins/plugin.js:196:46\n    at next (native)\n    at step (/usr/share/kibana/src/server/plugins/plugin.js:25:191)\n    at /usr/share/kibana/src/server/plugins/plugin.js:25:361"}}
```
As you can see in the error message... Not really sure where to start looking yet. It loads properly if it doesn't connect to Elasticsearch, and it loads if Searchguard is not installed, but it can connect to Elasticsearch.

**Comment by jochenkressin** [comment]
Which Browser and OS are you using?

**Comment by mikn** [comment]
This is an error I receive in the console when kibana is starting up, so
this is an error from nodejs... If that helps any.

**Comment by mikn** [comment]
So, I was running node 6.10, when I switched to 8.9.2 it seems to be working. Thank you.

**Comment by mikn** [comment]
Ah, no. Tracking it down, it seems to trigger when I disable basicauth.
```json
{"type":"log","@timestamp":"2017-12-09T05:07:11Z","tags":["status","plugin:kibana@6.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-12-09T05:07:11Z","tags":["status","plugin:elasticsearch@6.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-12-09T05:07:11Z","tags":["status","plugin:searchguard@6.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Search Guard HTTP Basic Authentication is disabled.","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-12-09T05:07:11Z","tags":["status","plugin:searchguard@6.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - Search Guard multitenancy disabled","prevState":"yellow","prevMsg":"Search Guard HTTP Basic Authentication is disabled."}
{"type":"log","@timestamp":"2017-12-09T05:07:11Z","tags":["status","plugin:searchguard@6.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - Search Guard copy JWT params disabled","prevState":"yellow","prevMsg":"Search Guard multitenancy disabled"}
{"type":"log","@timestamp":"2017-12-09T05:07:12Z","tags":["fatal"],"pid":1,"level":"fatal","message":"Unknown handler: async","error":{"message":"Unknown handler: async","name":"Error","stack":"Error: Unknown handler: async\n    at Object.exports.assert (/usr/share/kibana/node_modules/hoek/lib/index.js:736:11)\n    at Object.exports.defaults (/usr/share/kibana/node_modules/hapi/lib/handler.js:108:14)\n    at new module.exports.internals.Route (/usr/share/kibana/node_modules/hapi/lib/route.js:60:37)\n    at module.exports.internals.Connection.internals.Connection._addRoute (/usr/share/kibana/node_modules/hapi/lib/connection.js:387:19)\n    at module.exports.internals.Connection.internals.Connection._route (/usr/share/kibana/node_modules/hapi/lib/connection.js:379:18)\n    at module.exports.internals.Plugin.internals.Plugin._apply (/usr/share/kibana/node_modules/hapi/lib/plugin.js:572:14)\n    at module.exports.internals.Plugin.internals.Plugin.route (/usr/share/kibana/node_modules/hapi/lib/plugin.js:542:10)\n    at exports.default (/usr/share/kibana/plugins/searchguard/lib/configuration/routes/routes.js:21:10)\n    at ScopedPlugin.init [as externalInit] (/usr/share/kibana/plugins/searchguard/index.js:241:13)\n    at ScopedPlugin.tryCatcher (/usr/share/kibana/node_modules/bluebird/js/main/util.js:26:23)\n    at Promise.attempt.Promise.try (/usr/share/kibana/node_modules/bluebird/js/main/method.js:30:24)\n    at /usr/share/kibana/src/server/plugins/plugin.js:196:46\n    at Generator.next (<anonymous>)\n    at step (/usr/share/kibana/src/server/plugins/plugin.js:25:191)\n    at /usr/share/kibana/src/server/plugins/plugin.js:25:361\n    at <anonymous>"}}
```
This is still the nodejs logs from starting up the kibana service.

**Comment by jochenkressin** [comment]
Thanks, will check.

**Comment by jochenkressin** [comment]
Confirmed as a bug

**Comment by jochenkressin** [comment]
will be fixed in next release

**Comment by jochenkressin** [comment]
Fixed in 6.1.0-8

**Comment by berglh** [comment]
Glad to find it wasn't just me 😂
